src/main/java/org/sagebionetworks/bridge/reporter/worker/RetentionReportGenerator.java

```diff
@@ -0,0 +1,97 @@
+package org.sagebionetworks.bridge.reporter.worker;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.joda.time.DateTime;
+import org.joda.time.Days;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+
+import org.sagebionetworks.bridge.reporter.helper.BridgeHelper;
+import org.sagebionetworks.bridge.reporter.request.ReportType;
+import org.sagebionetworks.bridge.rest.model.AccountStatus;
+import org.sagebionetworks.bridge.rest.model.ActivityEventList;
+import org.sagebionetworks.bridge.rest.model.RequestInfo;
+import org.sagebionetworks.bridge.rest.model.SharingScope;
+import org.sagebionetworks.bridge.rest.model.StudyParticipant;
+
+import com.google.common.collect.HashMultiset;
+import com.google.common.collect.Multiset;
+
+/**
+ * Generate a report of signups by account statuses.
+ *
+ */
+@Component
+public class RetentionReportGenerator implements ReportGenerator {
+    
+    private BridgeHelper bridgeHelper;
+    
+    @Autowired
+    @Qualifier("ReporterHelper")
+    public final void setBridgeHelper(BridgeHelper bridgeHelper) {
+        this.bridgeHelper = bridgeHelper;
+    }
+    
+    @Override
+    public Report generate(BridgeReporterRequest request, String studyId) throws IOException {
+    	
+        DateTime startDate = request.getStartDateTime();
+        DateTime endDate = request.getEndDateTime();
+        String scheduler = request.getScheduler();
+        ReportType scheduleType = request.getScheduleType();
+        String reportId = scheduler + scheduleType.getSuffix();
+        
+        List<StudyParticipant> participants = bridgeHelper.getParticipantsForStudy(studyId, startDate, endDate);
```

Comment:
I'm a little worried about loading all participants and their request info objects or a study with many users, like Cardiovascular, which has 63,000 users. Iterating through the users and processing each in turn might be better.